### DISCUSSION TITLE
nvenc support?

### DISCUSSION THREAD

**Discussion by ohjoij-ys** [nagadomi/nunif#207] [Q&A] [answered]
Is there a possibility to support nvidia nvenc to accelerate encoding? or it is in cli?

**Comment by nagadomi** [comment, marked as answer]
1. pyav binary does not have h264_nvenc by default, so you need to build it yourself
2. I tried h264_nvenc before, but nothing improved.

So I decided not to support it.
The encoding process is parallelized and typically video encoding is faster than stereo generation, so processing time will not be improved.

**Reply by ohjoij-ys** [reply]
I found that my cpu was fully loaded when processing videos (13600k), and the gpu occupancy rate was only more than 80% (rtx3080), so I thought that when nvenc was used, the load of cpu encoding could be transferred to nvenc, but if it didn't actually work, then it would be meaningless. Thank you for your answer.

**Comment by nagadomi** [comment]
On Ubuntu, you can use nvenc,
`pip install av --force-reinstall --no-binary av`
and specify `--video-codec h264_nvenc` or `--video-codec hevc_nvenc`option (`Video Codec` on GUI).
However, other encoding options are ignored.
https://github.com/nagadomi/nunif/blob/a8f5e9178ff69cfd61c1f4e3aa04603a1f2890ce/iw3/utils.py#L606

**Reply by ohjoij-ys** [reply]
Do you mean modifying the nunif/iw3/utils.py file to:
options=options if args.video_codec in {"libx264", "libx265", "hevc_nvenc"} else {},\
Will the nvenc options appear, or do I have to add --video-codec hevc_nvenc in the cli

Or is there a place to add other parameters in the GUI? For example, this one? Please forgive me if my question is very newbie

**Reply by ohjoij-ys** [reply]
![image](https://github.com/user-attachments/assets/cb603d77-b0ee-4b33-8a9f-917678ac24eb)

**Reply by nagadomi** [reply]
In GUI, you can use hevc_nvenc by entering `hevc_nvenc` in `VideoCodec` field.

For video encoding options, nvenc is not compatible with libx265/libx264 options.
I don't know about the details.

**Comment by nagadomi** [comment]
I added NVENC support on Ubuntu https://github.com/nagadomi/nunif/commit/44cf3275bd1629b812afd5ea426670e5904709e1

Install PyAV from source.
```
sudo apt-get install ffmpeg libavcodec-dev libavformat-dev libavdevice-dev libavutil-dev libavfilter-dev libswscale-dev libswresample-dev
pip3 install av --force-reinstall --no-binary av
```
If NVENC is enabled in PyAV, it will appear in VideoCodec combobox.
![nvenc](https://github.com/user-attachments/assets/4ec8b828-bd78-4143-80b8-5664a6682de9)

**Reply by ohjoij-ys** [reply]
thank you ! It works
